### ISSUE TITLE
Rate limit issues on pipeline

### ISSUE THREAD

**Issue by mauromorales** [kairos-io/kairos#2838]
I'm not sure if dockerhub reduced their rate limits or if somehow we are just calling it more than before but it's been quite challenging to finish the release because of this:

```
logbus | Setting organization "" and project ""
              +all-arm | BASE_IMAGE=opensuse/tumbleweed FAMILY=opensuse FLAVOR=opensuse FLAVOR_RELEASE=tumbleweed K3S_VERSION=1.28.9 MODEL=rpi4 VARIANT=standard
              +all-arm | --> FROM +base
           alpine:3.19 | --> Load metadata alpine:3.19 linux/arm64
  Warning: you are not logged into registry-1.docker.io, you may experience rate-limitting when pulling images
              +all-arm | apply build +base: earthfile2llb for +base: Earthfile:2:0 apply FROM alpine:3.19: resolve image config for alpine:3.19: failed to copy: httpReadSeeker: failed open: unexpected status code https://registry-1.docker.io/v2/library/alpine/manifests/sha[25](https://github.com/kairos-io/kairos/actions/runs/10621879917/job/29463430008#step:12:27)6:95c16745f100f44cf9a0939fd3f357905f845f8b6fa7d0cde0e88c9764060185: 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
              +all-arm | in		github.com/kairos-io/kairos:v3.1.2+base
  
  Help: DockerHub responded with a rate limit error. This is usually because you are not logged in.
  You can login using the command:
    docker login
```

**Comment by jimmykarily** [comment]
Maybe we can login?

https://contribute.cncf.io/resources/project-services/hosted-tools/

**Comment by jimmykarily** [comment]
https://cncfservicedesk.atlassian.net/servicedesk/customer/portal/1/CNCFSD-2440?created=true (private link)

**Comment by jimmykarily** [comment]
We are supposed to apply through here: https://www.docker.com/community/open-source/application/
which requires us to have an org. To create an org [you normally have to pay](https://hub.docker.com/billing/core/purchase?type=organization) but they suggest this "workaround": https://docs.docker.com/admin/convert-account/#convert-a-community-account-into-an-organization

In other words, iiuc we are supposed to create a "kairos" user, convert that to an org account (hoping no payment will be required), then apply through the linked page. We can then ask the CNCF people (through the jira ticket) to push it further.

We are not going to push any images to dockerhub, we only need to be able to pull upstream distro images with better pull limits. I don't see why we need to just through hoops to create an org which we don't need...

I wonder if we should look at alternatives (e.g. use some registry mirror of our own to bybass the limits).

**Comment by mauromorales** [comment]
mirror would be ideal but not sure how complex that will be

**Comment by jimmykarily** [comment]
I followed their instructions and created an account which I converted to an org. Then I tried to apply through that form but I got these 2 errors:

![image](https://github.com/user-attachments/assets/babe362a-224a-40b8-a752-91bdc98c21d5)
![image](https://github.com/user-attachments/assets/8492c6ad-00e3-4f95-8229-0a7311ac313e)

I tried to woraround the second one by creating a "test" repository and then it complained about: 
![image](https://github.com/user-attachments/assets/8bd8b683-9f43-4030-a215-75ee15213ea3)

For the first error I don't even know what their problem is (the url is definitely right).

It seems that it's a lot of effort towards the wrong direction here. I'll update the jira ticket the close the org and all. We should better look for a way to avoid the limits (registry mirror or something).
